algorithms/matrix/matrix_characteristics.py

```diff
@@ -0,0 +1,55 @@
+# Class about simple matrix characteristics
+class Matrix(object):
+    def __init__(self, matrix):
+        self.rows = len(matrix)
+        self.columns = len(matrix[0])
+        self.matrix = matrix
+
+    def isLowertriangular(self):
+        matrix = self.matrix 
+        for i in range(0, len(matrix)): 
+            for j in range(i + 1, len(matrix)): 
+                if(matrix[i][j] != 0):  
+                        return False
+        return True
+
+    def isUppertriangular(self): 
+        matrix = self.matrix 
+        for i in range(1, len(matrix)): 
+            for j in range(0, i): 
+                if(matrix[i][j] != 0):  
+                        return False
+        return True
+
+    def isIdentity(self):
```

Comment:
The identity matrix is defined as a square matrix that has 1's along the main diagonal and 0's for all other entries.

You can check if ```isLowerTriangular```, ```isUpperTriangular``` and then if the values on the diagonal are 1's.

I think your code for the following input ```[[1, 1, 1, 1], [2, 1, 2, 2], [0, 0, 1, 0], [0, 0, 0, 1]]``` returns true, and the output should be false.